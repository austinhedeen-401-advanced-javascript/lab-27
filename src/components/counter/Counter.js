import React from 'react';

class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = { count: 0 };
  }

  handleUp = () => {
    const count = this.state.count + 1;
    this.updateCounter(count);
  };

  handleDown = () => {
    const count = this.state.count - 1;
    this.updateCounter(count);
  };

  updateCounter(count) {
    let polarity = '';
    if (count > 0) {
      polarity = 'positive';
    } else if (count < 0) {
      polarity = 'negative';
    }
    this.setState({ count, polarity });
  }

  render() {
    const classes = ['count', this.state.polarity].join(' ');
    return (
      <section className="counter">
        <button className="decrement" onClick={this.handleDown}>-</button>
        <span className={classes}>{this.state.count}</span>
        <button className="increment" onClick={this.handleUp}>+</button>
      </section>
    );
  }
}

export default Counter;
