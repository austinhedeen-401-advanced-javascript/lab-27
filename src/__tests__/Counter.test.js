import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '../components/counter/Counter';


describe('<Counter />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Counter />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('decrement button works as expected', () => {
    const app = mount(<Counter />);
    const countSpan = app.find('.counter span');
    const decrementButton = app.find('.decrement');

    expect(app.state('count')).toEqual(0);
    expect(countSpan.text()).toEqual('0');

    decrementButton.simulate('click');
    expect(app.state('count')).toEqual(-1);
    expect(countSpan.text()).toEqual('-1');

    decrementButton.simulate('click');
    expect(app.state('count')).toEqual(-2);
    expect(countSpan.text()).toEqual('-2');

    app.unmount();
  });

  test('increment button works as expected', () => {
    const app = mount(<Counter />);
    const countSpan = app.find('.counter span');
    const incrementButton = app.find('.increment');

    expect(app.state('count')).toEqual(0);
    expect(countSpan.text()).toEqual('0');

    incrementButton.simulate('click');
    expect(app.state('count')).toEqual(1);
    expect(countSpan.text()).toEqual('1');

    incrementButton.simulate('click');
    expect(app.state('count')).toEqual(2);
    expect(countSpan.text()).toEqual('2');

    app.unmount();
  });
});
