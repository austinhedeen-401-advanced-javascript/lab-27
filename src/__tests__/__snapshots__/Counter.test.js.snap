// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<section
  className="counter"
>
  <button
    className="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="count "
  >
    0
  </span>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</section>
`;
